src/main/java/com/bobocode/petros/container/DependencyDefinition.java

```diff
@@ -26,54 +29,74 @@ public DependencyDefinition() {
-                dependency = Class.forName(qualifiedName).getConstructor().newInstance();
-            } catch (Exception e) {
-                LOG.error("No default constructor found for class {}", qualifiedName);
-                throw new NoDefaultConstructorException(qualifiedName);
-            }
-        }
-        return dependency;
-    }
-
+    /**
+     * Return true in case if Dependency is declared inside configuration class that is marked with ConfigClass annotation
+     */
     public boolean isConfigClassDependency() {
         return configClassDependency;
     }
 
+    /**
+     * Set the configClassDependency flag to true in case if Dependency is declared inside configuration class that is marked with ConfigClass annotation
+     */
     public void setConfigClassDependency(boolean configClassDependency) {
         this.configClassDependency = configClassDependency;
     }
 
+    /**
+     * Return the name of the Dependency that is described by current DependencyDefinition
+     */
     public String getName() {
         return name;
     }
 
+    /**
+     * Set the name of the Dependency that is described by current DependencyDefinition
+     */
     public void setName(String name) {
         this.name = name;
     }
 
+    /**
+     * Return the Dependency qualified name that consists of the package that the Dependency class originated from and its name
+     */
     public String getQualifiedName() {
         return qualifiedName;
     }
 
+    /**
+     * Set the Dependency qualified name that consists of the package that the Dependency class originated from and its name
+     */
     public void setQualifiedName(String qualifiedName) {
         this.qualifiedName = qualifiedName;
     }
 
-    public String getInjectedDependencyMethodName() {
+    /**
+     * Return the name of the method that constructs Dependency described by current DependencyDefinition
+     *
+     * @return null in case if Dependency is not configuration
+     */
+    public String getConfigDependencyMethodName() {
         return injectedDependencyMethodName;
     }
 
-    public void setInjectedDependencyMethodName(String injectedDependencyMethodName) {
+    /**
+     * Set the name of the method that constructs Dependency described by current DependencyDefinition
+     */
+    public void setConfigDependencyMethodName(String injectedDependencyMethodName) {
         this.injectedDependencyMethodName = injectedDependencyMethodName;
     }
 
+    /**
+     * Add DependencyDefinition that Dependency described by current DependencyDefinition depends on
+     */
     public void addInjectedDependencyDefinition(DependencyDefinition dependencyQualifiedName) {
         injectedDependencyDefinitions.add(dependencyQualifiedName);
     }
 
+    /**
+     * Return a collection of all DependencyDefinitions that Dependency described by current DependencyDefinition depends on
+     */
     public Collection<DependencyDefinition> getDependencyDefinitions() {
```

Comment:
Please use annotation @return to describe return value

Reply:
In such cases, @return statement is omitted since it will duplicate the description.